.github/workflows/ci.yml

```
@@ -73,32 +33,76 @@ jobs:
-          docker logout
-      - name: Run Rubocop linting
+          BRANCH_REF="$(echo '${{ github.ref }}' | sed -E 's/[^A-Za-z0-9]+/-/g')"
+          echo ::set-env name=BRANCH_REF::$(BRANCH_REF)
+      - name: Log in to the Docker registry
         run: |
-          docker run --rm docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core/${{ matrix.suite }} && bundle exec rubocop . -c ../.rubocop.yml"
-      - name: Run ${{ matrix.suite }} tests with rspec
+          if [ -n "$DOCKER_USERNAME" ] && [ -n "$DOCKER_PASSWORD" ]; then
+            echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+          else
+            echo "No Docker credentials, skipping login"
+          fi
+      - name: Pull Docker base images & warm Docker cache
         run: |
-          docker run --rm docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }} bash -c "cd /home/dependabot/dependabot-core/${{ matrix.suite }} && bundle exec rspec spec"
-
-  npm-and-yarn-js:
-    name: npm and yarn js
-    runs-on: ubuntu-latest
-    needs: ci-image
-    steps:
-      - name: Pull image from packages
+          docker pull ubuntu:18.04
+          docker pull "dependabot/dependabot-core-ci:core--$BRANCH_REF" ||
+            docker pull "dependabot/dependabot-core:latest"
+          docker pull "dependabot/dependabot-core-ci:ci--$BRANCH_REF" ||
+            "dependabot/dependabot-core-ci:latest" || true
+      - name: Build dependabot-core image
+        run: |
+          docker build \
+            -t "dependabot/dependabot-core:latest" \
+            -t "dependabot/dependabot-core-ci:core--$BRANCH_REF" \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ubuntu:18.04 \
+            --cache-from "dependabot/dependabot-core-ci:core--$BRANCH_REF" \
+            --cache-from "dependabot/dependabot-core:latest" \
+            .
+      - name: Push core image to packages
         run: |
-          docker login docker.pkg.github.com -u x -p ${{ secrets.GITHUB_TOKEN }}
-          docker pull "docker.pkg.github.com/${{ github.repository }}/dependabot-core-ci:${{ github.sha }}"
-          docker logout
-      - name: Run js linting
+          if [ -n "$DOCKER_USERNAME" ] && [ -n "$DOCKER_PASSWORD" ]; then
+            docker push "dependabot/dependabot-core-ci:core--$BRANCH_REF"
+          else
+            echo "No Docker credentials, skipping push"
+          fi
+      - name: Build dependabot-core-ci image
+        run: |
+          rm .dockerignore
+          docker build \
+            -t "dependabot/dependabot-core-ci:latest" \
+            -t "dependabot/dependabot-core-ci:ci--$BRANCH_REF" \
+            -f Dockerfile.ci \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            --cache-from ubuntu:18.04 \
+            --cache-from "dependabot/dependabot-core:latest" \
```

```suggestion
            --cache-from "dependabot/dependabot-core-ci:latest" \
```

I think?  I'm extrapolating from :64 and :69.